services/jetbrains/jetbrains-base.js

```diff
@@ -1,11 +1,32 @@
 'use strict'
 
 const { BaseXmlService, NotFound } = require('..')
+const { parseJson } = require('../../core/base-service/json')
 
+/*
+JetBrains is a bit awkward. Sometimes we want to call an XML API
+and sometimes we want to call a JSON API so we need a mongrel base class.
+When the legacy IntelliJ (XML) API is retired we can simplify all this and
+switch JetbrainsDownloads, JetbrainsRating and JetbrainsVersion to just
+inherit from BaseJsonService directly.
+*/
 module.exports = class JetbrainsBase extends BaseXmlService {
+  static _isLegacyPluginId(pluginId) {
+    return !pluginId.match(/^([0-9])+/)
+  }
+
+  static _cleanPluginId(pluginId) {
+    const match = pluginId.match(/^([0-9])+/)
+    if (match) {
+      return match[0]
+    }
+    return pluginId
+  }
```

Comment:
Do we know definitively that for any `1234-foo-bar` plugin value the leading numeric portion will always match the underlying plugin id?

I.e. could a valid long form be `1234-foo-bar` but the corresponding plugin id be `4567`?

Based on some admittedly anecdotal testing I've done that does seem to be true, but if there's any conclusive doc/material from JetBrains that'd be helpful (an inline comment may be beneficial one way or the other for posterity)

Reply:
There's nowhere it is documented explicitly, but given that these (for example) do the same thing:
https://plugins.jetbrains.com/plugins/list?pluginId=1347-scala
https://plugins.jetbrains.com/plugins/list?pluginId=1347foobar
I'm pretty confident that the string part is irrelevant.

Reply:
That's so strange on the JetBrains side (accepting any randomness after the id) :laughing: